django/template/base.py

```diff
@@ -197,6 +197,15 @@ def compile_nodelist(self):
         except Exception as e:
             if self.engine.debug:
                 e.template_debug = self.get_exception_info(e, e.token)
+            if self.origin.name != UNKNOWN_SOURCE:
+                try:
+                    raw_message = e.args[0]
+                except IndexError:
+                    pass
```

Comment:
Can't we catch `TemplateSyntaxError` instead of try/excepting `e.args` access? 🤔